contracts/overseer/src/contract.rs

```diff
@@ -287,6 +330,114 @@ pub fn update_whitelist(
     ]))
 }
 
+fn update_rate(old_rate: Decimal256, rate_change: Decimal256, direction: bool) -> Decimal256 {
+    if direction {
+        old_rate + rate_change
+    } else if old_rate > rate_change {
+        old_rate - rate_change
+    } else {
+        old_rate
+    }
+}
+
+fn update_deposit_rate(deps: DepsMut, env: Env, deposit_rate: Decimal256) -> StdResult<Decimal256> {
+    let dynrate_config: DynrateConfig = read_dynrate_config(deps.storage)?;
+    let dynrate_state: DynrateState = read_dynrate_state(deps.storage)?;
+    let mut config: Config = read_config(deps.storage)?;
+    let mut new_deposit_rate: Decimal256 = deposit_rate;
+
+    // retrieve interest buffer
+    let interest_buffer = query_balance(
+        deps.as_ref(),
+        env.contract.address.clone(),
+        config.stable_denom.to_string(),
+    )?;
+
+    // check whether its time to re-evaluate rate
+    if !dynrate_state.prev_yield_reserve.is_zero()
+        && !interest_buffer.is_zero()
+        && env.block.height >= dynrate_state.last_executed_height + dynrate_config.dyn_rate_epoch
+    {
+        // passed time from the last executed time
+        let blocks_count = Uint256::from(env.block.height - dynrate_state.last_executed_height);
+
+        // yield reserve amt
+        let yield_reserve = Decimal256::from_uint256(interest_buffer);
+
+        // direction of rate change
+        let mut yr_went_up = yield_reserve > dynrate_state.prev_yield_reserve;
+
+        // normalized change in yr during dyn_rate_epoch
+        let yield_reserve_change = (if yr_went_up {
+            yield_reserve - dynrate_state.prev_yield_reserve
+        } else {
+            dynrate_state.prev_yield_reserve - yield_reserve
+        }) / dynrate_state.prev_yield_reserve;
+
+        // yr change per block
+        let mut yield_reserve_change_pb =
+            yield_reserve_change / Decimal256::from_uint256(blocks_count);
+
+        // increase expectation pb
+        let increase_expectation_pb = dynrate_config.dyn_rate_yr_increase_expectation
+            / Decimal256::from_uint256(blocks_count);
+
+        // consider increase expectation
+        yield_reserve_change_pb = if !yr_went_up {
+            yield_reserve_change_pb + increase_expectation_pb
+        } else if yield_reserve_change_pb > increase_expectation_pb {
+            yield_reserve_change_pb - increase_expectation_pb
+        } else {
+            // sign switched here
+            yr_went_up = !yr_went_up;
+            increase_expectation_pb - yield_reserve_change_pb
+        };
+
+        // change exceeded rate threshold, need to update variable rate
+        // recalc values from confing to a per block
+        let _year = Decimal256::from_uint256(BLOCKS_PER_YEAR);
```

Comment:
is this used anywhere?

Reply:
Not used anymore/will be removed, as it was decided that both configuration options should be per epoch. Thank you!